### PR TITLE
update registering an account

### PR DESCRIPTION
**Pull request by timofeevmd**

### CODE REVIEW

File: src/guide/kotlin-java.md [hunk 1]
```diff
@@ -11,13 +11,16 @@ the unusual, and provide some instructions for creating your own Iroha
 We assume that you know how to create a new package and have basic
 understanding of the fundamental Kotlin code. Specifically, we shall assume
 that you know how to build and deploy your program on the target platforms.
-The Iroha 2 JVM-compatible SDKs are as much a work-in-progress as the rest
-of this guide, and significantly more so than the Rust library.
+To clone Iroha 2 JVM compatible SDKs, you can use https://github.com/hyperledger/iroha-java.
```
Comment (outoftardis) [inline src/guide/kotlin-java.md:14]:
```suggestion
To clone Iroha 2 JVM compatible SDKs, you can use [Iroha Java](https://github.com/hyperledger/iroha-java).
```

File: src/guide/kotlin-java.md [hunk 2]
```diff
@@ -103,22 +104,94 @@ Iroha makes extensive use of asynchronous programming (in Rust
 terminology), hence blocking is not necessarily the only mode of
 interaction with the Iroha 2 code.
 
-We have started by creating a mutable lazy-initialised client. This client
-is passed an instance of a domain registration box, which we get as a
-result of evaluating `registerDomain(domainName)`. Then the client is sent
-a transaction which consists of that one instruction. And that's it.
+In order to make sure that the raised peers work correctly, you can do a simple
```
Comment (outoftardis) [inline src/guide/kotlin-java.md:107]: running peers maybe?

File: src/guide/kotlin-java.md [hunk 3]
```diff
@@ -129,35 +202,56 @@ heap-allocated reference types are all called boxes).
...
+    sendTransaction.registerAccount(joe, listOf(joeKeyPair.public.toIrohaPublicKey()))
+        .also { println("ACCOUNT $joe CREATED") }
 
-Similarly to the previous example, we provide the instructions in the form
-of a test:
+    query.findAllAccounts()
+        .also { println("ALL ACCOUNTS: ${it.map { a -> a.id.asString() }}") }
+```
+
+And implement new method on class SendTransaction to your project.
```
Comment (outoftardis) [inline src/guide/kotlin-java.md:219]:
```suggestion
Then implement new method for class `SendTransaction` in your project.
```

File: src/guide/kotlin-java.md [hunk 4]
```diff
@@ -129,35 +202,56 @@ heap-allocated reference types are all called boxes).
...
-account for _white_rabbit@looking_glass_, she should provide his public
-key.
+has to have a key pair. 
 
-It is tempting to generate both the private and public keys at this time,
-but it isn't the brightest idea. Remember that _the white_rabbit_ trusts
-_you, alice@wonderland,_ to create an account for them in the domain
-_looking_glass_, **but doesn't want you to have access to that account
-after creation**.
+To register a new account add next lines on Main.kt
```
Comment (outoftardis) [inline src/guide/kotlin-java.md:207]:
```suggestion
To register a new account, add the following lines to `Main.kt`:
```

File: src/guide/kotlin-java.md [hunk 5]
```diff
@@ -103,22 +104,94 @@ Iroha makes extensive use of asynchronous programming (in Rust
...
-has to be done by an account. This is a special account that must already
-exist on the blockchain. You can ensure that point by reading through
-`genesis.json` and seeing that **_alice_** indeed has an account, with a
-public key. Furthermore, the account's public key must be included in the
-configuration. If either of these two is missing, you will not be able to
-register an account, and will be greeted by an exception of an appropriate
-type.
+:::
+
+To register a new domain add next lines on Main.kt
```
Comment (outoftardis) [inline src/guide/kotlin-java.md:152]:
```suggestion
To register a new domain, add the following lines to Main.kt:
```

File: src/guide/kotlin-java.md [hunk 6]
```diff
@@ -103,22 +104,94 @@ Iroha makes extensive use of asynchronous programming (in Rust
...
+
+```kotlin
+val sendTransaction = SendTransaction(peerUrl, telemetryUrl, client, admin, adminKeyPair)
+
+val domain = "domain_${System.currentTimeMillis()}"
+    sendTransaction.registerDomain(domain).also { println("DOMAIN $domain CREATED") }
+```
+
+And create new open class SendTransaction to your project.
```
Comment (outoftardis) [inline src/guide/kotlin-java.md:161]:
```suggestion
Then create new open class `SendTransaction` in your project:
```

File: src/guide/kotlin-java.md (hunk 2 above)
Comment (timofeevmd) [reply src/guide/kotlin-java.md:107]: oh, thank you, you are right

File: src/guide/kotlin-java.md [hunk 7]
```diff
@@ -77,24 +80,22 @@ and see how the `Iroha2Config` is implemented.
 
 <<<@/snippets/IrohaConfig.kotlin
 
-## 3. Registering a Domain
+## 3. Domain search and registration
 
-Registering a domain is one of the easier operations. The usual boilerplate
+Search Registering a domain is one of the easier operations. The usual boilerplate
```
Comment (outoftardis) [inline src/guide/kotlin-java.md:85]: search registered domains?

Comment (timofeevmd) [reply src/guide/kotlin-java.md:85]: Search and registering - sure

Comment (astrokov7) [reply src/guide/kotlin-java.md:85]:
```suggestion
Querying and Registering a domain are among the easier operations. The usual boilerplate
```

File: src/guide/kotlin-java.md [hunk 8]
```diff
@@ -77,24 +80,22 @@ and see how the `Iroha2Config` is implemented.
 
 <<<@/snippets/IrohaConfig.kotlin
 
-## 3. Registering a Domain
+## 3. Domain search and registration
```
Comment (astrokov7) [inline src/guide/kotlin-java.md:83]:
```suggestion
## 3. Querying and Registering Domains
```

File: src/guide/kotlin-java.md [hunk 9]
```diff
@@ -77,24 +80,22 @@ and see how the `Iroha2Config` is implemented.
 
 <<<@/snippets/IrohaConfig.kotlin
 
-## 3. Registering a Domain
+## 3. Querying and Registering Domains
 
-Registering a domain is one of the easier operations. The usual boilerplate
+Querying and Registering a domain are among the easier operations. The usual boilerplate
```
Comment (WRRicht3r) [inline src/guide/kotlin-java.md:85]:
```suggestion
Querying and Registering a domain are easier operations. The usual boilerplate
```

**Review by WRRicht3r — CHANGES_REQUESTED** [review]

**Review by WRRicht3r — APPROVED** [review]

**Review by 6r1d — APPROVED** [review]
LGTM

**Pull request merged**